### PR TITLE
[full-ci] Implement copy, cut, paste in ContextMenu | Fix nav icon flickering in lightmode

### PR DESCRIPTION
**Pull request by lookacat**
## Description
LocationPicker in Contextmenu is now replaced with clipboard actions (as already used with keyboard actions)
See https://github.com/owncloud/web/issues/6892

## What needs to be fixed in acceptance tests:
We've replaced the right-click contextmenu actions "copy" and "move", which opened the locationpicker, with clipboard actions. Now we have "cut", "copy" and "paste" instead and they should behave like e.g. in windows explorer.
Therefore we need to adjust how the tests copy or move files since the old way doesn't exist anymore.

## Screenshots
Copy/paste keyboard shortcuts in contextmenu
image
Cutting elements "blurs" relevant resources / Paste becomes visible when there is an item in clipboard
image
Paste here button becomes visible if there are items in clipboard
image
Clipboard can be cleared by clicking close on the paste button (which also removes the paste here button)
image

## Related Issue

- Works towards https://github.com/owncloud/web/issues/6892

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/26945/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/26945/tracing/file-alice-2022-7-21-09-50-17.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/26945/tracing/file-brian-2022-7-21-09-50-24.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/26945/tracing/folder-alice-2022-7-21-09-48-46.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/26945/tracing/folder-brian-2022-7-21-09-48-53.zip 


</p></details>

File: packages/web-app-files/src/mixins/actions/paste.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          name: 'paste',
+          icon: 'clipboard',
+          handler: this.$_paste_trigger,
+          label: () =>
+            this.$pgettext('Action in the files list row to initiate pasting resources', 'Paste'),
+          shortcut: this.getPasteShortcutString,
+          isEnabled: ({ resources }) => {
+            if (this.clipboardResources.length === 0) return false
+            if (
+              !isLocationSpacesActive(this.$router, 'files-spaces-personal') &&
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/actions/paste.js:30]: What about shares? That's the `files-spaces-share` spaces location.

File: packages/web-app-files/src/mixins/actions/paste.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { mapActions, mapMutations, mapGetters } from 'vuex'
+
+export default {
+  computed: {
+    ...mapGetters('Files', ['clipboardResources']),
+    isMacOs() {
+      return window.navigator.platform.match('Mac')
+    },
+    getPasteShortcutString() {
+      const modifierKey = this.isMacOs ? '⌘' : 'Ctrl'
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/actions/paste.js:15]: Needs to be translated. Please use it with translation comment:
```js
if (this.isMacOs) {
  return this.$pgettext('Keyboard shortcut for macOS for pasting files', '⌘ + V')
}
return this.$pgettext('Keyboard shortcut for non-macOS systems for pasting files', 'Ctrl + V')
```

**Review by kulmann — COMMENTED** [review]
I have mixed feelings about this PR. Feels like really good progress. But technically I don't like to see more and more in the vuex store... 🤔

**Comment by kulmann** [comment]
Had a glitch with what seems to be a prefix match issue. When I create two folders with names `test` and `test2` on the same level, then copy `test`, navigate into `test2` and try to paste I get an error message that I can't paste a folder into itself.

**Comment by lookacat** [comment]
@phil-davis @individual-it could someone from your team please take over required changes in the acceptance tests? :)

**Comment by SagarGi** [comment]
@lookacat @kulmann  Some findings with this PR

1. After copying a specific file and selecting a destination folder to copy we have no `cancel` options to cancel copy as previouly.

(screenshot from master build)

Screenshot from 2022-08-02 15-37-09

need to know if its a feature or the `X` works as cancel in here?

2. And also we cannot copy paste in public link resources



https://user-images.githubusercontent.com/46086950/182353445-e47348ea-a64d-46f9-9ac7-b488235ffd54.mp4

needs some information regarding this.

**Comment by kulmann** [comment]
> 1. After copying a specific file and selecting a destination folder to copy we have no `cancel` options to cancel copy as previouly.
> need to know if its a feature or the `X` works as cancel in here?

we don't need to test the cancel option anymore, as we don't navigate into a different view (location picker is obsolete with this PR). you just stay in the files list of any files related view. As a result you can delete tests that cancel the location picker.


> 2. And also we cannot copy paste in public link resources
> 
>  cannotcopyinpubliclink.mp4 
> 
> needs some information regarding this.

Needs to be fixed by a dev. Thanks for the pointer!

**Comment by SagarGi** [comment]
> @SagarGi
> 
> I added a fix for your second finding (pasting broken in public links) in [417f875](https://github.com/owncloud/web/commit/417f87569b28877ef2934e0b18a3b850dc23f1f6)
> 
> And in [6a9c79d](https://github.com/owncloud/web/commit/6a9c79dea80c0c5ef567367e2dc2758670069f24) I removed the location picker entirely, as it is not used anymore in the UI. There are some tests referencing the location picker. Some can be deleted (cancelling copy or move), but for some I'm not sure if you can rewrite them easily (anything that's successful copy/move e.g. from the personal page) or if the time would be better spent writing new ones in the e2e playwright tests and deleting them in the nightwatch test suite. What do you think?

@kulmann if location picker is entirely removed then tests related to it on (nightwatch) can be removed and time can be spent on e2e playwright for that as you say.

**Comment by SagarGi** [comment]
@kulmann i forgot to pull your changes from (https://github.com/owncloud/web/pull/7309#issuecomment-1209769720) to my local and force push it (my very bad). So i have added them with commit [1b8cb5b](https://github.com/owncloud/web/pull/7309/commits/1b8cb5bf547967ecb96ff1ebfbc848b6f98f6167) and [c541a9e](https://github.com/owncloud/web/pull/7309/commits/c541a9e2100af52dd73262e9c2832897983d308b)

**Comment by kulmann** [comment]
> @kulmann i forgot to pull your changes from ([#7309 (comment)](https://github.com/owncloud/web/pull/7309#issuecomment-1209769720)) to my local and force push it (my very bad). So i have added them with commit [1b8cb5b](https://github.com/owncloud/web/pull/7309/commits/1b8cb5bf547967ecb96ff1ebfbc848b6f98f6167) and [c541a9e](https://github.com/owncloud/web/pull/7309/commits/c541a9e2100af52dd73262e9c2832897983d308b)

No worries. Thanks for re-adding them. For future force pushes, just use `git push --force-with-lease`, that will abort the force push if the remote branch had changes in the meantime. Or just the shorthand `gpf` if you use https://ohmyz.sh

**Comment by kulmann** [comment]
> > @SagarGi
> > I added a fix for your second finding (pasting broken in public links) in [417f875](https://github.com/owncloud/web/commit/417f87569b28877ef2934e0b18a3b850dc23f1f6)
> > And in [6a9c79d](https://github.com/owncloud/web/commit/6a9c79dea80c0c5ef567367e2dc2758670069f24) I removed the location picker entirely, as it is not used anymore in the UI. There are some tests referencing the location picker. Some can be deleted (cancelling copy or move), but for some I'm not sure if you can rewrite them easily (anything that's successful copy/move e.g. from the personal page) or if the time would be better spent writing new ones in the e2e playwright tests and deleting them in the nightwatch test suite. What do you think?
> 
> @kulmann if location picker is entirely removed then tests related to it on (nightwatch) can be removed and time can be spent on e2e playwright for that as you say.

Cool. Thank you for already taking care! :heart: I have removed some more dead code in the nightwatch test suite and moved the `paste` action from the location picker page object (deleted now) to the files list page object. Hope the test suite becomes green now :-)

**Review by kulmann — APPROVED** [review]
Awesome! 😍 The UX without the location picker is so much better, love it.

**Comment by SagarGi** [comment]
@SwikritiT @saw-jan @kiranparajuli589 may be someone can review the testscode changes.

**Comment by kulmann** [comment]
> @SwikritiT @saw-jan @kiranparajuli589 may be someone can review the testscode changes.

I already did 😅 my review was for code and test-code. Looks all good to me 💪

**Comment by kulmann** [comment]
Temporarily converted back to draft because I want to get https://github.com/owncloud/web/pull/7425 merged first. That one already had too many rebases...

**Comment by SagarGi** [comment]
> > @SwikritiT @saw-jan @kiranparajuli589 may be someone can review the testscode changes.
> 
> I already did sweat_smile my review was for code and test-code. Looks all good to me muscle

Thanks for if in that case.

**Pull request merged**